### PR TITLE
[stable/stolon] Stolon ssl

### PR DESCRIPTION
**Pull request by jmvizcainoio**
#### What this PR does / why we need it:

This PR add support to enable ssl to postgresql via chart.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by jmvizcainoio** [comment]
@davidkarlsen @sameersbn Hi, any update about the PR ? Thanks

**Comment by zanhsieh** [comment]
@jmvizcainoio 
This chart will be hard to merge since there's no OWNERS file.

**Comment by jmvizcainoio** [comment]
> @jmvizcainoio
> This chart will be hard to merge since there's no OWNERS file.

Hi , there are another PR creating OWNERS [file](https://github.com/helm/charts/pull/16292) , i hope that someone move this and accept the PR.

@davidkarlsen @sameersbn  @rtluckie @lwolf  any update?

**Comment by zanhsieh** [comment]
@jmvizcainoio 
I suggest you search them on kubernetes slack channel.

**Comment by merqurio** [comment]
Is there anyway to help with the process ?

**Comment by lwolf** [comment]
I'll take a look tonight.

**Comment by merqurio** [comment]
Thanks @lwolf I really appreciate it. Happy to become a maintainer, may I open a PR and add myself so I can help you in the future ?

**Comment by lwolf** [comment]
@merqurio I guess you can create a PR, but the one about adding myself as maintainer is pending merge for 2 months now.
I'm thinking about hosting it myself again and adding it to the hub.

**Comment by zanhsieh** [comment]
@lwolf 
May I know which PR do you talk about? I only find this:
https://github.com/helm/charts/pull/16292
What I will try to do is, go to Kubernetes slack channel, find them, and direct msg them.

File: stable/stolon/values.yaml [hunk 1]
```diff
@@ -74,6 +74,17 @@ clusterSpec: {}
   # sleepInterval: 1s
   # maxStandbys: 5
 
+## Enable support ssl into postgres, you must specify the certs.
+## ref: https://www.postgresql.org/docs/10/ssl-tcp.html
+##
+tls:
+  enabled: false
+  rootCa: |-
```
Comment (lwolf) [inline stable/stolon/values.yaml:82]: I wonder, should it be always up to the user to create a secret, to not have private key in the values.yaml?

File: stable/stolon/README.md [hunk 2]
```diff
@@ -58,6 +58,10 @@ Kubernetes is the default store backend. `consul`, `etcdv2` or `etcdv3` can also
 | `job.autoCreateCluster`                 | Set to `false` to force-disable auto-cluster-creation which may clear pre-existing postgres db data | `true`  |
 | `job.autoUpdateClusterSpec`             | Set to `false` to force-disable auto-cluster-spec-update | `true`                                             |
 | `clusterSpec`                           | Stolon cluster spec [reference](https://github.com/sorintlab/stolon/blob/master/doc/cluster_spec.md) | `{}`   |
+| `tls.enabled`                           | Enable tls support to postgresql               | `false`                                                      |
+| `tls.rootCa`                            | Ca ceertificate                                | `""`                                                         |
```
Comment (lwolf) [inline stable/stolon/README.md]: typo `ceertificate`

File: stable/stolon/README.md [hunk 3]
```diff
@@ -58,6 +58,10 @@ Kubernetes is the default store backend. `consul`, `etcdv2` or `etcdv3` can also
...
+| `tls.serverCrt`                         | Server Cerfificate                             | `""`                                                         |
```
Comment (lwolf) [inline stable/stolon/README.md]: typo `Cerfificate`

File: stable/stolon/templates/hooks/update-cluster-spec-job.yaml [hunk 4]
```diff
@@ -42,5 +42,5 @@ spec:
             - --store-endpoints={{ .Values.store.endpoints }}
             {{- end }}
             - -p
-            - '{ {{- range $key, $value := .Values.clusterSpec }} {{ $key | quote }}: {{ if typeIs "string" $value }} {{ $value | quote }} {{ else }} {{ $value }} {{ end }}, {{- end }} "pgParameters": {{ toJson .Values.pgParameters }} }'
+            - '{ {{- range $key, $value := .Values.clusterSpec }} {{ $key | quote }}: {{ if typeIs "string" $value }} {{ $value | quote }} {{ else }} {{ $value }} {{ end }}, {{- end }} {{ if .Values.tls.enabled }}"pgParameters": {{- $pgParameters := .Values.pgParameters -}}{{ $all := set $pgParameters "ssl" "on" }}{{ $all := set $all "ssl_cert_file" "/certs/serverCrt.crt" }} {{ $all := set $all "ssl_key_file" "/certs/serverKey.key" }}{{ $all := set $all "ssl_ca_file" "/certs/rootCa.crt" }}{{ toJson $all }} {{ else }}"pgParameters": {{- $pgParameters := .Values.pgParameters -}}{{ $all := set $pgParameters "ssl" nil }}{{ $all := set $all "ssl_cert_file" nil }} {{ $all := set $all "ssl_key_file" nil }}{{ $all := set $all "ssl_ca_file" "null" }}{{ toJson $all }}{{ end}}}'
```
Comment (lwolf) [inline stable/stolon/templates/hooks/update-cluster-spec-job.yaml:45]: rendering with ssl enabled and certs provided seems fine, but I'm not sure how stolon will interpret `null`, `"null"` values. can we just omit this ssl block if ssl is disabled?
```
- '{ "initMode": "new", "pgParameters": {"ssl":null,"ssl_ca_file":"null","ssl_cert_file":null,"ssl_key_file":null}  }'
```

**Review by lwolf — CHANGES_REQUESTED** [review]
mostly lgtm, except for a few small comments

**Comment by lwolf** [comment]
@zanhsieh yes I was referring to the #16292

File: stable/stolon/templates/hooks/update-cluster-spec-job.yaml (hunk 4 above)
Comment (jmvizcainoio) [reply stable/stolon/templates/hooks/update-cluster-spec-job.yaml:45]: Thanks for your comments. 

This template is only executed when perform an upgrade. The null is to remove the ssl config if was configured before and do you want disable for any reason. When analyzing the null value, stolon will be removed from the configuration. If the values ​​do not exist, nothing happens.

File: stable/stolon/values.yaml (hunk 1 above)
Comment (jmvizcainoio) [reply stable/stolon/values.yaml:82]: Hello, thanks for your comments. The user can choose, secrets can be created before and specify the name of the secret using

`` suggestion
createdSecret: existing-secret-name-a-postgresql
``

or type the certificate in values.yaml.
I want to assume that the user encrypts sensible information in the values.yaml file, for example, using helm secrets

File: stable/stolon/README.md (hunk 2 above)
Comment (jmvizcainoio) [reply stable/stolon/README.md]: Thanks for the comment, fixed.

File: stable/stolon/README.md (hunk 3 above)
Comment (jmvizcainoio) [reply stable/stolon/README.md]: Thanks for the comment, fixed.

**Comment by merqurio** [comment]
Seems all tests passed, @sameersbn could you please check this PR ? Thanks a lot for your time !

**Comment by zanhsieh** [comment]
@jmvizcainoio 
Could you resolve the conflicting files please?

stable/stolon/Chart.yaml

**Comment by jmvizcainoio** [comment]
> @jmvizcainoio
> Could you resolve the conflicting files please?
> 
> stable/stolon/Chart.yaml

Sure, version bumped. Thanks

**Review by lwolf — APPROVED** [review]

File: stable/stolon/Chart.yaml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: stolon
-version: 1.3.0
+version: 1.3.1
```
Comment (sameersbn) [inline stable/stolon/Chart.yaml]: @jmvizcainoio please bump the minor chart version

**Review by sameersbn — CHANGES_REQUESTED** [review]

**Review by sameersbn — APPROVED** [review]
/lgtm

**Pull request merged**